test/parallel/test-sqlite-backup.mjs

```diff
@@ -0,0 +1,217 @@
+    const database = makeSourceDb();
+
+    t.assert.throws(() => {
+      database.backup('hello.db', {
+        source: 42
+      });
+    }, {
+      code: 'ERR_INVALID_ARG_TYPE',
+      message: 'The "options.source" argument must be a string.'
+    });
+
+    t.assert.throws(() => {
+      database.backup('hello.db', {
+        target: 42
+      });
+    }, {
+      code: 'ERR_INVALID_ARG_TYPE',
+      message: 'The "options.target" argument must be a string.'
+    });
+
+    t.assert.throws(() => {
+      database.backup('hello.db', {
+        rate: 'invalid'
+      });
+    }, {
+      code: 'ERR_INVALID_ARG_TYPE',
+      message: 'The "options.rate" argument must be an integer.'
+    });
+
+    t.assert.throws(() => {
+      database.backup('hello.db', {
+        progress: 'invalid'
+      });
+    }, {
+      code: 'ERR_INVALID_ARG_TYPE',
+      message: 'The "options.progress" argument must be a function.'
+    });
+  });
+});
+
+test('database backup', async (t) => {
+  const progressFn = t.mock.fn();
+  const database = makeSourceDb();
+  const destDb = nextDb();
+
+  await database.backup(destDb, {
+    rate: 1,
+    progress: progressFn,
+  });
+
+  const backup = new DatabaseSync(destDb);
+  const rows = backup.prepare('SELECT * FROM data').all();
+
+  // The source database has two pages - using the default page size -,
+  // so the progress function should be called once (the last call is not made since
+  // the promise resolves)
+  t.assert.strictEqual(progressFn.mock.calls.length, 1);
+  t.assert.deepStrictEqual(progressFn.mock.calls[0].arguments, [{ totalPages: 2, remainingPages: 1 }]);
+  t.assert.deepStrictEqual(rows, [
+    { __proto__: null, key: 1, value: 'value-1' },
+    { __proto__: null, key: 2, value: 'value-2' },
+  ]);
+});
+
+test('database backup in a single call', async (t) => {
+  const progressFn = t.mock.fn();
+  const database = makeSourceDb();
+  const destDb = nextDb();
+
+  // Let rate to be default (100) to backup in a single call
+  await database.backup(destDb, {
+    progress: progressFn,
+  });
+
+  const backup = new DatabaseSync(destDb);
+  const rows = backup.prepare('SELECT * FROM data').all();
+
+  t.assert.strictEqual(progressFn.mock.calls.length, 0);
+  t.assert.deepStrictEqual(rows, [
+    { __proto__: null, key: 1, value: 'value-1' },
+    { __proto__: null, key: 2, value: 'value-2' },
+  ]);
+});
+
+test('throws exception when trying to start backup from a closed database', async (t) => {
+  t.assert.throws(() => {
+    const database = new DatabaseSync(':memory:');
+
+    database.close();
+
+    database.backup('backup.db');
+  }, {
+    code: 'ERR_INVALID_STATE',
+    message: 'database is not open'
+  });
+});
+
+test('database backup fails when dest file is not writable', (t) => {
+  const readonlyDestDb = nextDb();
+  writeFileSync(readonlyDestDb, '', { mode: 0o444 });
+
+  const database = makeSourceDb();
+
+  t.assert.rejects(async () => {
```

Comment:
Shouldn't `rejects()` calls here and in other places be `await`-ed?

Reply:
Makes sense. Will fix this.

Reply:
Done